src/io/mod.rs

```diff
@@ -48,3 +47,357 @@ mod stdin;
+            InvalidData,
+            /// The I/O operation's timeout expired, causing it to be canceled.
+            TimedOut,
+            /// An error returned when an operation could not be completed because a
+            /// call to [`write`] returned [`Ok(0)`].
+            ///
+            /// This typically means that an operation could only succeed if it wrote a
+            /// particular number of bytes but only a smaller number of bytes could be
+            /// written.
+            ///
+            /// [`write`]: ../io/trait.Write.html#tymethod.write
+            /// [`Ok(0)`]: ../io/type.Result.html
+            WriteZero,
+            /// This operation was interrupted.
+            ///
+            /// Interrupted operations can typically be retried.
+            Interrupted,
+            /// Any I/O error not part of this list.
+            Other,
+            /// An error returned when an operation could not be completed because an
+            /// "end of file" was reached prematurely.
+            ///
+            /// This typically means that an operation could only succeed if it read a
+            /// particular number of bytes but only a smaller number of bytes could be
+            /// read.
+            UnexpectedEof,
+        }
+    } else {
+        #[doc(inline)]
+        pub use std::io::ErrorKind;
+    }
+}
+
+cfg_if! {
+    if #[cfg(feature = "docs")] {
+        /// A specialized [`Result`](../result/enum.Result.html) type for I/O
+        /// operations.
+        ///
+        /// This type is broadly used across [`std::io`] for any operation which may
+        /// produce an error.
+        ///
+        /// This typedef is generally used to avoid writing out [`io::Error`] directly and
+        /// is otherwise a direct mapping to [`Result`].
+        ///
+        /// While usual Rust style is to import types directly, aliases of [`Result`]
+        /// often are not, to make it easier to distinguish between them. [`Result`] is
+        /// generally assumed to be [`std::result::Result`][`Result`], and so users of this alias
+        /// will generally use `io::Result` instead of shadowing the prelude's import
+        /// of [`std::result::Result`][`Result`].
+        ///
+        /// [`std::io`]: ../io/index.html
+        /// [`io::Error`]: ../io/struct.Error.html
+        /// [`Result`]: ../result/enum.Result.html
+        ///
+        /// # Examples
+        ///
+        /// A convenience function that bubbles an `io::Result` to its caller:
+        ///
+        /// ```
+        /// use std::io;
+        ///
+        /// fn get_string() -> io::Result<String> {
+        ///     let mut buffer = String::new();
+        ///
+        ///     io::stdin().read_line(&mut buffer)?;
+        ///
+        ///     Ok(buffer)
+        /// }
+        /// ```
+        pub type Result<T> = std::result::Result<T, std::io::Error>;
+    } else {
+        #[doc(inline)]
+        pub use std::io::Result;
+    }
+}
+
+cfg_if! {
```

Comment:
We don't need to add this section for the `Write` trait because our `Write` trait is different from the one in `std` :)